server/src/main/java/io/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -91,7 +92,7 @@ public Void withHandle(Handle handle) throws Exception
                     getSupervisorsTable()
                 )
             )
-                  .bind("id", String.format("%s_%s", id, version))
+                  .bind("id", String.format("%s_%s_%s", id, version, UUIDUtils.generateUuid()))
```

Comment:
Hm, having the UUID as part of the version wouldn't really help much, would it? *If* in the real world this actually happened, the "newest" spec between two with identical timestamps would be random, regardless if version included the UUID or not. In MySQL, calling max(version) when there's multiple rows matching still returns a single row.